### PR TITLE
Fix No Answer Inference

### PR DESCRIPTION
**Pull request by paneer**
## Summary

Use Null Conditional and Null Coalescing operators to suggest `No Answer`

Fixes dotnet/docs#14542

### CODE REVIEW

File: csharp/NullableIntroduction/NullableIntroduction/SurveyResponse.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ public class SurveyResponse
 
         // <SnippetSurveyStatus>
         public bool AnsweredSurvey => surveyResponses != null;
-        public string Answer(int index) => surveyResponses.GetValueOrDefault(index, "No answer")!;
+        public string Answer(int index) => surveyResponses?.GetValueOrDefault(index) ?? "No answer";
```
Comment (Thraka) [inline csharp/NullableIntroduction/NullableIntroduction/SurveyResponse.cs]: I think this should be

```suggestion
        public string Answer(int index) => surveyResponses?[index] ?? "No answer";
```

Comment (paneer) [reply csharp/NullableIntroduction/NullableIntroduction/SurveyResponse.cs]: @Thraka, index op on a dictionary even with a null conditional operator may result in a runtime `KeyNotFoundException`. Obviously, one could check for `ContainsKey` but the resulting code would be pretty verbose:
```suggestion
        public string Answer(int index) => surveyResponses?.GetValueOrDefault(index) ?? "No answer";
```
I simply left the extension method as it makes the code readable.

Comment (BillWagner) [reply csharp/NullableIntroduction/NullableIntroduction/SurveyResponse.cs]: This is a good discussion.  @paneer is correct, the extension method is needed.

I'll mark this conversation resolved.

Comment (Thraka) [reply csharp/NullableIntroduction/NullableIntroduction/SurveyResponse.cs]: Ahh yes duh on me!

**Pull request merged**